Lib/test/test_importlib/extension/test_loader.py

```diff
@@ -29,6 +30,24 @@ def test_load_module_API(self):
         with self.assertRaises(ImportError):
             self.load_module('XXX')
 
+    def test_type_error(self):
+        with self.assertRaises(TypeError) as e:
+            other = self.machinery.ExtensionFileLoader(
+                fullname=util.EXTENSIONS.name,
+                name=util.EXTENSIONS.name,
+                path=util.EXTENSIONS.file_path)
+
+            self.assertEqual(e.exception.message,
+                         'Either name or fullname needs to be used')
+
+    def test_name_deprecation(self):
+        with support.check_warnings(("the 'name' parameter is deprecated; use "
```

Comment:
`support.check_warnings()` predates `assertWarns`. It would be better if we use `assertWarns`.

```suggestion
        with self.assertWarns(DeprecationWarning) as cm:
            other = self.machinery.ExtensionFileLoader(
                name=util.EXTENSIONS.name,
                path=util.EXTENSIONS.file_path,
            )
        self.assertEqual(
            str(cm.warning),
            "the 'name' parameter is deprecated; use 'fullname' instead"
        )
```